reactor-netty-http-brave/src/main/java/reactor/netty/http/brave/TracingHttpClientDecorator.java

```diff
@@ -0,0 +1,313 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.brave;
+
+import brave.Span;
+import brave.SpanCustomizer;
+import brave.http.HttpClientHandler;
+import brave.http.HttpClientRequest;
+import brave.http.HttpClientResponse;
+import brave.http.HttpTracing;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+import reactor.core.publisher.Mono;
+import reactor.netty.Connection;
+import reactor.netty.http.client.HttpClient;
+import reactor.util.annotation.Nullable;
+import reactor.util.context.ContextView;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is based on
+ * https://github.com/spring-cloud/spring-cloud-sleuth/blob/d969aa7b549e9c82729e16dc7d846654e32ddef9/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/HttpClientBeanPostProcessor.java
+ */
+final class TracingHttpClientDecorator {
+
+	HttpClient decorate(HttpClient client) {
+		TracingDoOnResponse onResponse = new TracingDoOnResponse(handler, uriMapping);
+		return client.doOnRequest(new TracingDoOnRequest(handler, uriMapping))
+		             .doOnResponse(onResponse)
+		             .doOnRedirect(onResponse)
+		             .doOnError(new TracingDoOnRequestError(handler, uriMapping),
+		                        new TracingDoOnResponseError(handler, uriMapping))
+		             .mapConnect(new TracingMapConnect(currentTraceContext));
+	}
+
+	final CurrentTraceContext currentTraceContext;
+	final HttpClientHandler<HttpClientRequest, HttpClientResponse> handler;
+	final Function<String, String> uriMapping;
+
+	TracingHttpClientDecorator(HttpTracing httpTracing, Function<String, String> uriMapping) {
+		requireNonNull(httpTracing, "httpTracing");
+		this.currentTraceContext = httpTracing.tracing().currentTraceContext();
+		this.handler = HttpClientHandler.create(httpTracing);
+		this.uriMapping = requireNonNull(uriMapping, "uriMapping");
+	}
+
+	static abstract class AbstractTracingDoOnHandler {
+
+		final HttpClientHandler<HttpClientRequest, HttpClientResponse> handler;
+
+		AbstractTracingDoOnHandler(HttpClientHandler<HttpClientRequest, HttpClientResponse> handler) {
+			this.handler = handler;
+		}
+
+		void handleReceive(ContextView contextView, HttpClientResponse response) {
```

Comment:
I would either fully embrace the mirroring of Brave's `HttpClientHandler` by swapping the arguments (Brave's `handleReceive` is response then `Span`, here it would be response then `ContextView`) or rename it enough that it isn't ambiguous this is a rnetty method and not an implementation of a `Brave` interface

Reply:
fixed